### ISSUE TITLE
The `tailscale_tailnet_key` resource for non-reusable keys doesn't work

### ISSUE THREAD

**Issue by michaelbeaumont** [davidsbond/terraform-provider-tailscale#59]
**Describe the bug**
After applying a plan involving a non-reusable key, any further operations are impossible because of an unexpected 404 error when trying to query the API for the key.

**To Reproduce**
Apply a plan with a `"tailscale_tailnet_key"` resource with `reusable = false` via `terraform apply` and try to delete the resources using `terraform destroy` and get something like the following error, preventing any further action.

```
│ Error: Failed to fetch key
│
│   with tailscale_tailnet_key.gcp,
│   on main.tf line 16, in resource "tailscale_tailnet_key" "gcp":
│   16: resource "tailscale_tailnet_key" "gcp" {
│
│ 404 page not found (404)
```

**Expected behaviour**
No errors. I don't think it should be necessary to query the tailscale API if the key is present in the state file.

**Comment by davidsbond** [comment]
Hey @michaelbeaumont thanks for raising this, I hadn't thought about it. I'm wondering what the best thing to do for non-reusable keys would be. 

Since reading the keys from the API is useless, I think the appropriate fix for now is just making no API calls on reads as you suggest. However I wonder if it makes your HCL files messy with resources that the state file says exist but actually don't.

**Comment by davidsbond** [comment]
Also a side note for fixing this, the response when deleting keys should not error if they get a 404, since this could be a non-reusable key that has been used.

**Comment by michaelbeaumont** [comment]
> However I wonder if it makes your HCL files messy with resources that the state file says exist but actually don't.

Depends on what you mean by messy. Because we have it in the state, we can always read it and destroying it is a noop, so can we kind of pretend it continues to exist?

The only issue I see with non-reusable is what to do if the key's used by more than one machine, but that would be an issue even if the key continued to exist in the API. Maybe there's a way to enforce a single-use with HCL/terraform.

I'm not sure what other providers do in cases like this, couldn't find a comparable example with a quick search.

**Comment by engineeringsys** [comment]
I also ran into this with one-time keys.  I think the best route is to avoid reading the key from the API.  As @michaelbeaumont mentioned, finding an example of this to see how they bail out of it has proven challenging.

**Comment by davidsbond** [comment]
I should have some time to take a look at this in a few days time. Otherwise PRs are welcome and I'll review them as soon as I can

**Comment by davidsbond** [comment]
@engineeringsys @michaelbeaumont, I've just published 0.7.0 which show be available within the terraform registry shortly. Please let me know if this resolves your issue. If not, comment here and I'll reopen this.
